common/src/main/java/org/opensearch/ml/common/conversational/ConversationMeta.java

```diff
@@ -0,0 +1,176 @@
+     * @return new conversationMeta object
+     * @throws IOException if you're reading from a stream without a conversationMeta in it
+     */
+    public static ConversationMeta fromStream(StreamInput in) throws IOException {
+        String id = in.readString();
+        Instant created = in.readInstant();
+        Instant lastHit = in.readInstant();
+        int numInteractions = in.readInt();
+        String name = in.readString();
+        String user = in.readOptionalString();
+        return new ConversationMeta(id, created, lastHit, numInteractions, name, user);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(id);
+        out.writeInstant(created);
+        out.writeInstant(lastHit);
+        out.writeInt(numInteractions);
+        out.writeString(name);
+        out.writeOptionalString(user);
+    }
+
+    /**
+     * Hit this conversationMeta at this time, increasing the converation length
+     * @param hitTime the Instant when the new interaction was created
+     * @return this conversationMeta object (fields updated)
+     */
+    public ConversationMeta hit(Instant hitTime) {
+        this.lastHit = hitTime;
+        this.numInteractions++;
+        return this;
+    }
+
+    
+    /**
+     * Convert this conversationMeta object into an IndexRequest so it can be indexed
+     * @param index the index to send this conversation to. Should usually be .conversational-meta
+     * @return the IndexRequest for the client to send
+     */
+    public IndexRequest toIndexRequest(String index) {
+        IndexRequest request = new IndexRequest(index);
+        return request.id(this.id).source(
+            ConversationalIndexConstants.META_CREATED_FIELD, this.created,
+            ConversationalIndexConstants.META_ENDED_FIELD, this.lastHit,
+            ConversationalIndexConstants.META_LENGTH_FIELD, this.numInteractions,
+            ConversationalIndexConstants.META_NAME_FIELD, this.name
+        );
+    }
+
+    @Override
+    public String toString() {
+        return "{id=" + id
+            + ", name=" + name
+            + ", length=" + numInteractions
+            + ", created=" + created.toString()
+            + ", lastHit=" + lastHit.toString()
+            + ", user=" + user
+            + "}";
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContentObject.Params params) throws IOException {
+        builder.startObject();
+        builder.field(ActionConstants.CONVERSATION_ID_FIELD, this.id);
+        builder.field(ConversationalIndexConstants.META_CREATED_FIELD, this.created);
+        builder.field(ConversationalIndexConstants.META_ENDED_FIELD, this.lastHit);
+        builder.field(ConversationalIndexConstants.META_LENGTH_FIELD, this.numInteractions);
+        builder.field(ConversationalIndexConstants.META_NAME_FIELD, this.name);
+        builder.field(ConversationalIndexConstants.USER_FIELD, this.user);
```

Comment:
Check field is null or not, same suggestion applies to other places.
```suggestion
if (this.user != null) {
        builder.field(ConversationalIndexConstants.USER_FIELD, this.user);
}

```